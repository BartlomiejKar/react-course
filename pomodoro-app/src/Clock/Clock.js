import React from 'react';
import "./Clock.css"
function Clock({
    className = "",
    hours = 0,
    minutes = 0,
    seconds = 0,
    miliseconds = 0
}) {
    return (
        <h2 className={"Clock " + className}>
            Pozosta≈Ço {Hours(hours)}:{Minutes(minutes)}:{Seconds(seconds)}:
{Miliseconds(miliseconds)}
        </h2>
    );
}
function Hours(hours) {
    const TimeHours = hours < 0 ? `00` : hours < 10 ? `0${hours}` : hours;
    return `${TimeHours}`;
}

function Minutes(minutes) {
    const TimeMinutes =
        minutes < 0 ? `00` : minutes < 10 ? `0${minutes}` : minutes;
    return `${TimeMinutes}`;
}
function Seconds(seconds) {
    const TimeSeconds =
        seconds < 0 ? `00` : seconds < 10 ? `0${seconds}` : seconds;
    return `${TimeSeconds}`;
}
function Miliseconds(miliseconds) {
    const TimeMiliseconds =
        miliseconds < 0
            ? `00`
            : miliseconds < 10
                ? `0${miliseconds}`
                : miliseconds;
    return `${TimeMiliseconds}`;
}

export default Clock;