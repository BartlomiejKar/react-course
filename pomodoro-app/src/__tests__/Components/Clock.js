import React from 'react';
import Clock from "../../Clock/Clock"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
var clockRenderer = null;
var root = null
describe("<Clock/>", () => {
    describe("when gives minutes and seconds (DOM)", () => {
        beforeEach(() => {
            root = document.createElement("div");
            ReactDOM.render(
                <Clock minutes={20} seconds={10} />, root
            )
        })
        it("renders properly", () => {
            expect(root.childNodes[0].nodeName).toEqual("H2")
            expect(root.childNodes[0].className).toMatch(/Clock/)
            expect(root.childNodes[0].textContent).toMatch(/20:10/)
        })
    })
    it("render h2 element", () => {
        expect(root.childNodes[0].nodeName).toEqual("H2")
    })
    it("set className", () => {
        expect(root.childNodes[0].className).toMatch(/Clock/)
    })
    it("render time properly", () => {
        expect(root.childNodes[0].textContent).toMatch(/20:10/)
    })

    describe("when gives minutes and seconds(react-test-renderer)", () => {
        beforeEach(() => {
            clockRenderer = renderer.create(
                <Clock minutes={20} seconds={10} />
            );
        })
        it("renders properly", () => {
            console.log(clockRenderer.toJSON())
            expect(clockRenderer.toJSON()).toMatchSnapshot()
        })
    })
    it("render h2 element", () => {
        expect(clockRenderer.toJSON().type).toEqual("h2")
    })
    it("set className", () => {
        expect(clockRenderer.toJSON().props).toMatchObject({ "className": expect.stringMatching(/Clock/) })
    })
    it("render time properly", () => {
        expect(clockRenderer.toJSON().children).toEqual(expect.arrayContaining(["20", "10"]))
    })
})