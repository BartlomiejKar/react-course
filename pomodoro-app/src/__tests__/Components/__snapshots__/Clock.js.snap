// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock/> when gives minutes and seconds(react-test-renderer) renders properly 1`] = `
<h2
  className="Clock "
>
  Pozosta≈Ço 
  00
  :
  20
  :
  10
  :
  00
</h2>
`;
